source/common/upstream/upstream_impl.h

```diff
@@ -305,6 +325,10 @@ class HostImpl : public HostDescriptionImpl,
 
   std::atomic<uint32_t> health_flags_{};
   std::atomic<uint32_t> weight_;
+  // TODO(wbpcode): should we store the EDS health status to health_flags_ to get unified status or
```

Comment:
I think it might ultimately be cleaner to just eliminate `Host::Health` and instead expose only the `HealthStatus` value, which would represent the final state after all inputs are applied (EDS health status, active health checking, etc).

Reply:
I think this is fair, but maybe use better names, since `health` and `healthStatus` are both seemingly the same thing. Maybe rename the existing one to `coarseHealth` or something with similar intent (naming is hard).

Reply:
naming is hard. 🤣 I think `coarseHealth` is good enough. I will rename it.

Reply:
As per my other comment, I wonder how many places we could just switch to the new health status. If it's too risky to do it all at once, I think it's fine to do what you have in this PR, and maybe under runtime flag protection migrate the existing callsites?